app/client/components/SearchBoxWithFilter.jsx

```diff
@@ -89,14 +89,19 @@ class SearchBoxWithFilter extends React.Component {
 
     const proccess_attribute = (name, value) => {
       if (value !== undefined && value !== null) {
+        
         if (attribute_dict[name] === undefined) {
           // This means that the attribute name has not been seen at all, which means we can just create array
-          attribute_dict[name] = [value]
+          attribute_dict[name] = {
+            values: [value],
+            type: typeof(value) == 'number' ? USER_FILTER_TYPE.INT_RANGE : USER_FILTER_TYPE.DISCRETE
+          }
+          
         } else {
           // Attribute name has been seen, check if attribute VALUE has been seen
-          if (attribute_dict[name].indexOf(value) === -1) {
+          if (attribute_dict[name].values.indexOf(value) === -1) {
```

Comment:
we could probably upgrade this to using a set. Saves doing indexOf which is slow (comparatively)